hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioInboundPipeline.java

```diff
@@ -106,6 +106,11 @@ public long lastReadTimeMillis() {
     @Override
     void process() throws Exception {
         processCount.inc();
+        if (receiveBuffer == null) {
```

Comment:
I don't think this is good. Process should not be called when the receive buffer is null.

I don't like checks for exceptional cases on the fast path.